src/java/org/apache/cassandra/cache/ChunkCache.java

```diff
@@ -226,15 +231,21 @@ public CachingRebufferer(ChunkReader file)
         @Override
         public Buffer rebuffer(long position)
         {
+            int spin = 0;
             try
             {
                 long pageAlignedPos = position & alignmentMask;
                 Buffer buf;
-                do
-                    buf = cache.get(new Key(source, pageAlignedPos)).reference();
-                while (buf == null);
-
-                return buf;
+                Key key = new Key(source, pageAlignedPos);
+                while (true)
+                {
+                    buf = cache.get(key).reference();
+                    if (buf != null)
+                        return buf;
+
+                    if (++spin == 1024)
+                        logger.error("Spinning for {}", key);
```

Comment:
The error should be a little more descriptive, e.g. "Could not acquire a reference for {} after 1000 attempts. This is likely due to too small chunk cache size causing contention." plus a code comment saying this may be caused by reference counting errors.

It may be better to throw an exception than to let this continue indefinitely.